pkg/sdkserver/sdkserver_test.go

```diff
@@ -709,17 +710,31 @@ func TestSDKServerWatchGameServerFeatureSDKWatchSendOnExecute(t *testing.T) {
 	assert.Nil(t, waitConnectedStreamCount(sc, 1))
 	assert.Equal(t, stream, sc.connectedStreams[0])
 
+	// modify for 2nd event in watch stream
+	fixture.Status.State = agonesv1.GameServerStateAllocated
```

Comment:
@roberthbailey @akremsa - I reworked this test to check for:
1. We get what we expect in events on watch and modification
1. That we don't get too many events
1. We can't go into an infinite loop

PTAL, let me know what you think

Reply:
Nice update! I like the way you've updated a select section.

Reply:
I also suggested to use state other than Ready.